staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -242,3 +248,54 @@ func generateKey(length int) (key []byte, err error) {
 
 	return key, nil
 }
+
+func validateEncryptedObject(o *kmstypes.EncryptedObject) error {
+	if o == nil {
+		return fmt.Errorf("encrypted object is nil")
+	}
+	if o.EncryptedDEK == nil {
+		return fmt.Errorf("encrypted dek is nil")
+	}
+	if o.EncryptedData == nil {
+		return fmt.Errorf("encrypted data is nil")
+	}
```

Comment:
`[]byte{}` is also invalid:

```suggestion
	if len(o.EncryptedDEK) == 0 {
		return fmt.Errorf("encrypted dek is empty")
	}
	if len(o.EncryptedData) == 0 {
		return fmt.Errorf("encrypted data is empty")
	}
```